platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/component.py

```diff
@@ -0,0 +1,394 @@
+
+        if not self.__is_dict(module):
+            self.__parser_module_fail("dictionary is expected: key={}".format(self.MODULE_KEY))
+
+        if not module:
+            self.__parser_module_fail("dictionary is empty: key={}".format(self.MODULE_KEY))
+
+        for key, value in module.items():
+            if not self.__is_dict(value):
+                self.__parser_module_fail("dictionary is expected: key={}".format(key))
+
+            if not value:
+                self.__parser_module_fail("dictionary is empty: key={}".format(key))
+
+            if self.COMPONENT_KEY not in value:
+                self.__parser_module_fail("\"{}\" key hasn't been found".format(self.COMPONENT_KEY))
+
+            if len(value) != 1:
+                self.__parser_module_fail("unexpected number of records: key={}".format(key))
+
+            self.__module_component_map[key] = OrderedDict()
+            self.__parse_component_section(key, value[self.COMPONENT_KEY], True)
+
+    def get_components_list(self):
+        self.__component_list = []
+        for key, value in self.__chassis_component_map[self.__bf_model].items():
+            self.__component_list.append(key)
+        
+        return self.__component_list
+
+    def get_chassis_component_map(self):
+        return self.__chassis_component_map
+
+    def get_module_component_map(self):
+        return self.__module_component_map
+
+    def parse_platform_components(self,  platform_components_path):
+        with open(platform_components_path) as platform_components:
+            data = json.load(platform_components)
+
+            if not self.__is_dict(data):
+                self.__parser_platform_fail("dictionary is expected: key=root")
+
+            if not data:
+                self.__parser_platform_fail("dictionary is empty: key=root")
+
+            if self.CHASSIS_KEY not in data:
+                self.__parser_platform_fail("\"{}\" key hasn't been found".format(self.CHASSIS_KEY))
+
+            if not self.__is_modular_chassis:
+                if len(data) != 1:
+                    self.__parser_platform_fail("unexpected number of records: key=root")
+
+            self.__parse_chassis_section(data[self.CHASSIS_KEY])
+
+            if self.__is_modular_chassis:
+                if self.MODULE_KEY not in data:
+                    self.__parser_platform_fail("\"{}\" key hasn't been found".format(self.MODULE_KEY))
+
+                if len(data) != 2:
+                    self.__parser_platform_fail("unexpected number of records: key=root")
+
+                self.__parse_module_section(data[self.MODULE_KEY])
+
+    chassis_component_map = property(fget=get_chassis_component_map)
+    module_component_map = property(fget=get_module_component_map)
+
+class Components(ComponentBase):
+    """BFN Montara Platform-specific Component class"""
+
+    #bf_model = get_model()
+    bpcp = BFPlatformComponentsParser(is_modular_chassis=False,  bf_model="Wedge100BF-32X-O-AC-F-BF")
```

Comment:
it should be generic plugin for Montara/Maveriks/Newport
please use sonic_py_common library to get platform type
e.g., https://github.com/Azure/sonic-buildimage/blob/3bbe562bcaeb3e16fec86f27c185d30b5b44cf6e/platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/fan_drawer.py#L87